tritium-lib/src/main/java/com/palantir/tritium/io/ForwardingOutputStream.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.io;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Objects;
+
+abstract class ForwardingOutputStream extends FilterOutputStream {
```

Comment:
I left these `Forwarding*Stream` as package private for now. If one really wants, Apache commons-io's [`Proxy*Stream`](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/output/ProxyOutputStream.html) provides similar framework